### ISSUE TITLE
gcr.io/k8s-prow: some of the images don't seem to get pushed after move to dedicated repo

### ISSUE THREAD

**Issue by dhiller** [kubernetes-sigs/prow#121]
# tl;dr
it seems that some prow images aren't getting pushed to `gcr.io/k8s-prow` any more, 
i.e. we noticed that the `commenter` image is missing for tag `v20240418-4c9d8ca12`. Same goes for tag `v20240410-4be743f3e`

# long story
We (the https://github.com/kubevirt GitHub org) run our own prow instance https://prow.ci.kubevirt.io/. We have automation in place that updates the prow image tags in our infrastructure repository https://github.com/kubevirt/project-infra for KubeVirt CI in periodic intervals.

Since the move of prow into it's dedicated repository https://github.com/kubernetes/test-infra/issues/31728 some of the images don't get pushed any more.

The below is a `grep` of the prow images we are using currently, combined with a listing of the tags available from the image repository.

```bash
$ for image in $(git grep -oE 'gcr.io/k8s-prow/[a-z0-9_]+' -- ./github | \
    sed 's#.*:#docker://#' | sort -u); \
  do \
      echo "$image has $(skopeo list-tags $image | grep '"v20240418-4c9d8ca12' | wc -l) images with tag v20240418-4c9d8ca12"; \
  done
docker://gcr.io/k8s-prow/branchprotector has 1 images with tag v20240418-4c9d8ca12
docker://gcr.io/k8s-prow/checkconfig has 1 images with tag v20240418-4c9d8ca12
docker://gcr.io/k8s-prow/cherrypicker has 1 images with tag v20240418-4c9d8ca12
docker://gcr.io/k8s-prow/clonerefs has 4 images with tag v20240418-4c9d8ca12
docker://gcr.io/k8s-prow/commenter has 0 images with tag v20240418-4c9d8ca12
docker://gcr.io/k8s-prow/configurator has 0 images with tag v20240418-4c9d8ca12
docker://gcr.io/k8s-prow/crier has 1 images with tag v20240418-4c9d8ca12
docker://gcr.io/k8s-prow/deck has 1 images with tag v20240418-4c9d8ca12
docker://gcr.io/k8s-prow/entrypoint has 4 images with tag v20240418-4c9d8ca12
docker://gcr.io/k8s-prow/exporter has 1 images with tag v20240418-4c9d8ca12
docker://gcr.io/k8s-prow/ghproxy has 1 images with tag v20240418-4c9d8ca12
docker://gcr.io/k8s-prow/grandmatriarch has 1 images with tag v20240418-4c9d8ca12
docker://gcr.io/k8s-prow/hook has 1 images with tag v20240418-4c9d8ca12
docker://gcr.io/k8s-prow/horologium has 1 images with tag v20240418-4c9d8ca12
docker://gcr.io/k8s-prow/initupload has 4 images with tag v20240418-4c9d8ca12
docker://gcr.io/k8s-prow/label_sync has 0 images with tag v20240418-4c9d8ca12
docker://gcr.io/k8s-prow/needs has 0 images with tag v20240418-4c9d8ca12
docker://gcr.io/k8s-prow/peribolos has 1 images with tag v20240418-4c9d8ca12
docker://gcr.io/k8s-prow/pipeline has 1 images with tag v20240418-4c9d8ca12
docker://gcr.io/k8s-prow/prow has 0 images with tag v20240418-4c9d8ca12
docker://gcr.io/k8s-prow/sidecar has 4 images with tag v20240418-4c9d8ca12
docker://gcr.io/k8s-prow/sinker has 1 images with tag v20240418-4c9d8ca12
docker://gcr.io/k8s-prow/status has 0 images with tag v20240418-4c9d8ca12
docker://gcr.io/k8s-prow/tide has 1 images with tag v20240418-4c9d8ca12
docker://gcr.io/k8s-prow/tot has 1 images with tag v20240418-4c9d8ca12
```

**Note: this list might not be complete wrt all the prow images pushed though**

/cc @timwangmusic @cjwagner

Affected PRs that we had to revert:
* https://github.com/kubevirt/project-infra/pull/3340
* https://github.com/kubevirt/project-infra/pull/3354

**Comment by dhiller** [comment]
FYI @brianmcarey

**Comment by jihoon-seo** [comment]
Hello @dhiller

For anyone's information:
It seems that the components with 0 images from the log above reside in `kubernetes/test-infra` repo, which leads to
1. those components are technically not a Prow component, and
2. we might (& should) check whether the postsubmit job of `kubernetes/test-infra` repo which builds & pushes images is working correctly.

WDYT? :)
(Please bear in mind that I might be wrong.)

**Comment by dhiller** [comment]
> Hello @dhiller
> 
> It seems that the components with 0 images from the log above reside in `kubernetes/test-infra` repo, which leads to
> 
> 1. those components are technically not a Prow component, and
> 2. we might (& should) check whether the postsubmit job of `kubernetes/test-infra` repo which builds & pushes images is working correctly.
> 
> WDYT? :) (Please bear in mind that I might be wrong.)

I see in k/test-infra the following comment

```
# The following images are published to gcr.io/k8s-prow even though they are not Prow images for legacy purposes
```
([source](https://github.com/kubernetes/test-infra/blob/56c4bbfde0fd8d7f0135c680919ecd105fcd9d72/.k8s-prow-images.yaml#L1))

so I think that the image publishing should still work as before.

**Comment by dhiller** [comment]
The above meaning that I believe _some_ publishing should still happen. I wouldn't have a problem with fetching another tag and splitting the update logic, since I guess it would be hard to maintain a single tag for both builds.

Maybe I am just looking at the wrong place?

**Comment by dhiller** [comment]
Hm, I see the prowimagebuilder has been moved to prow repo also, so it might not be as easy as re-adding the prow job that has been removed with commit [455adb0d6ccdda9f33fef08ab01c3bb062c31753](https://github.com/kubernetes/test-infra/commit/455adb0)

**Comment by cjwagner** [comment]
This change was part of https://github.com/kubernetes/test-infra/issues/31728, specifically, it sounds like you are missing this addition to your autobump config: https://github.com/kubernetes/test-infra/issues/31728#issuecomment-2048438546

Non-prow images that were historically published to gcr.io/k8s-prow due to convenience/laziness can no longer be published with consistent image tags as the prow images, because they are now published from different repos. The images are still publshed to gcr.io/k8s-prow for backwards compatibility, but the non-prow images are no longer expected (and never needed) to have consistent tags.

I'll close this as WAI, but let me know if you have any questions.
